common/source/Util.cpp

```diff
@@ -34,6 +39,21 @@ std::string exec(const char *cmd) {
   return result;
 }
 
+int readFile(std::string fileName, int arraySize, unsigned char* array)
```

Comment:
make fileName a const reference, you shouldn't be changing it. 

Please also check to see if variables in the code are snake case, I'm pretty sure they are.

Reply:
array size should also be const.